### ISSUE TITLE
Plugin only supports older/weaker cipher suits

### ISSUE THREAD

**Issue by olavaukan** [OctopusDeploy/Octopus-TeamCity#65]
So I spent about a full day trying to figure out why this plugin wasn't working until I made the conclusion that for some reason it only supports a handful of older/weaker SSL cipher suits and therefore the Build Agent wasn't able to connect to my Build Server, which is behind an reverse proxy Nginx server.

My TeamCity build server is accessible through an Nginx revers proxy with SSL termination and it's been "hardened" to only allow stronger cipher suits:

```
ssl_protocols TLSv1.2 TLSv1.3;
ssl_ciphers ECDHE-RSA-AES256-GCM-SHA512:DHE-RSA-AES256-GCM-SHA512:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES256-GCM-SHA384;
```
I added an 'OctopusDeploy: Build information' step to my build definition, but it kept failing with the following error:

```
[Step 25/26] java.lang.Error: Failed to connect to https://build.mydomain.com/httpAuth/app/rest/builds/id:1355: Received fatal alert: protocol_version
	at org.jetbrains.teamcity.rest.TeamCityInstanceImpl$service$3.handleError(implementation.kt:73)
	at org.jetbrains.teamcity.rest.TeamCityInstanceImpl$service$3.handleError(implementation.kt:51)
	at retrofit.RestAdapter$RestHandler.invoke(RestAdapter.java:242)
	at com.sun.proxy.$Proxy78.build(Unknown Source)
	at org.jetbrains.teamcity.rest.TeamCityInstanceImpl.build(implementation.kt:79)
	at octopus.teamcity.agent.OctopusBuildInformationBuildProcess.createCommand(OctopusBuildInformationBuildProcess.java:75)
	at octopus.teamcity.agent.OctopusBuildProcess.start(OctopusBuildProcess.java:53)
	at jetbrains.buildServer.agent.impl.buildStages.runnerStages.start.CallRunnerStage.doBuildStage(CallRunnerStage.java:60)
	at jetbrains.buildServer.agent.impl.buildStages.RunnerStagesExecutor$1.callStage(RunnerStagesExecutor.java:27)
	at jetbrains.buildServer.agent.impl.buildStages.RunnerStagesExecutor$1.callStage(RunnerStagesExecutor.java:18)
	at jetbrains.buildServer.agent.impl.buildStages.StagesExecutor.callRunStage(StagesExecutor.java:76)
	at jetbrains.buildServer.agent.impl.buildStages.StagesExecutor.doStages(StagesExecutor.java:35)
	at jetbrains.buildServer.agent.impl.buildStages.RunnerStagesExecutor.doStages(RunnerStagesExecutor.java:18)
	at jetbrains.buildServer.agent.impl.buildStages.startStages.steps.RunnerContextExecutor.callRunnerStages(RunnerContextExecutor.java:43)
	at jetbrains.buildServer.agent.impl.buildStages.startStages.steps.StepExecutor.processNextStep(StepExecutor.java:25)
	at jetbrains.buildServer.agent.impl.buildStages.startStages.steps.ForEachBuildRunnerStage.executeRunnerStep(ForEachBuildRunnerStage.java:139)
	at jetbrains.buildServer.agent.impl.buildStages.startStages.steps.ForEachBuildRunnerStage.runStep(ForEachBuildRunnerStage.java:124)
	at jetbrains.buildServer.agent.impl.buildStages.startStages.steps.ForEachBuildRunnerStage.executeBuildRunners(ForEachBuildRunnerStage.java:84)
	at jetbrains.buildServer.agent.impl.buildStages.startStages.steps.ForEachBuildRunnerStage.doBuildStage(ForEachBuildRunnerStage.java:45)
	at jetbrains.buildServer.agent.impl.buildStages.BuildStagesExecutor$1.callStage(BuildStagesExecutor.java:33)
	at jetbrains.buildServer.agent.impl.buildStages.BuildStagesExecutor$1.callStage(BuildStagesExecutor.java:24)
	at jetbrains.buildServer.agent.impl.buildStages.StagesExecutor.callRunStage(StagesExecutor.java:76)
	at jetbrains.buildServer.agent.impl.buildStages.StagesExecutor.doStages(StagesExecutor.java:35)
	at jetbrains.buildServer.agent.impl.buildStages.BuildStagesExecutor.doStages(BuildStagesExecutor.java:24)
	at jetbrains.buildServer.agent.impl.BuildRunActionImpl.doStages(BuildRunActionImpl.java:81)
	at jetbrains.buildServer.agent.impl.BuildRunActionImpl.runBuild(BuildRunActionImpl.java:57)
	at jetbrains.buildServer.agent.impl.BuildAgentImpl.doActualBuild(BuildAgentImpl.java:333)
	at jetbrains.buildServer.agent.impl.BuildAgentImpl.access$200(BuildAgentImpl.java:58)
	at jetbrains.buildServer.agent.impl.BuildAgentImpl$1.run(BuildAgentImpl.java:288)
	at java.lang.Thread.run(Thread.java:748)
Caused by: retrofit.RetrofitError: Received fatal alert: protocol_version
	at retrofit.RetrofitError.networkError(RetrofitError.java:27)
	at retrofit.RestAdapter$RestHandler.invokeRequest(RestAdapter.java:395)
	at retrofit.RestAdapter$RestHandler.invoke(RestAdapter.java:240)
	... 27 more
Caused by: javax.net.ssl.SSLHandshakeException: Received fatal alert: protocol_version
	at sun.security.ssl.Alert.createSSLException(Alert.java:131)
	at sun.security.ssl.Alert.createSSLException(Alert.java:117)
	at sun.security.ssl.TransportContext.fatal(TransportContext.java:311)
	at sun.security.ssl.Alert$AlertConsumer.consume(Alert.java:293)
	at sun.security.ssl.TransportContext.dispatch(TransportContext.java:185)
	at sun.security.ssl.SSLTransport.decode(SSLTransport.java:152)
	at sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1383)
	at sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1291)
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:435)
	at okhttp3.internal.io.RealConnection.connectTls(RealConnection.java:188)
	at okhttp3.internal.io.RealConnection.connectSocket(RealConnection.java:145)
	at okhttp3.internal.io.RealConnection.connect(RealConnection.java:108)
	at okhttp3.internal.http.StreamAllocation.findConnection(StreamAllocation.java:188)
	at okhttp3.internal.http.StreamAllocation.findHealthyConnection(StreamAllocation.java:127)
	at okhttp3.internal.http.StreamAllocation.newStream(StreamAllocation.java:97)
	at okhttp3.internal.http.HttpEngine.connect(HttpEngine.java:289)
	at okhttp3.internal.http.HttpEngine.sendRequest(HttpEngine.java:241)
	at okhttp3.RealCall.getResponse(RealCall.java:240)
	at okhttp3.RealCall$ApplicationInterceptorChain.proceed(RealCall.java:198)
	at org.jetbrains.teamcity.rest.RetryInterceptor.intercept(implementation.kt:37)
	at okhttp3.RealCall$ApplicationInterceptorChain.proceed(RealCall.java:187)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:160)
	at okhttp3.RealCall.execute(RealCall.java:57)
	at com.jakewharton.retrofit.Ok3Client.execute(Ok3Client.java:40)
	at retrofit.RestAdapter$RestHandler.invokeRequest(RestAdapter.java:326)
	... 28 more
	Suppressed: javax.net.ssl.SSLHandshakeException: Received fatal alert: handshake_failure protocol_version
		... 53 more
```
The TeamCity build agent was configured to talk to the TeamCity server using the public HTTPS address as this was needed for it to generate metadata for OctopusDeploy with the correct URLs back to TeamCity. The problem went away if I configured the build agent to talk to the build server using the internal HTTP address.

After a while I was able to get debug-logging to work and found the following in the logs:

```
javax.net.ssl|FINE|4B|TC: 17:15:37 Build {id=1362, buildTypeId='BuildApplicationInfrastructureImages', buildTypeInternalId='bt10'}|2021-07-20 17:23:37.242 BST|ClientHello.java:575|Produced ClientHello handshake message ( 
"ClientHello": { 
  "client version"      : "TLSv1", 
  "random"              : "5C A9 9C 7F 58 56 18 78 50 7D EB 9B 80 66 73 E2 14 BF 29 1E 70 65 8C A8 22 98 28 29 86 10 A2 9C", 
  "session id"          : "", 
  "cipher suites"       : "[TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA(0xC00A), TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA(0xC009), TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA(0xC013), TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA(0xC014), TLS_DHE_RSA_WITH_AES_128_CBC_SHA(0x0033), TLS_DHE_RSA_WITH_AES_256_CBC_SHA(0x0039), TLS_RSA_WITH_AES_128_CBC_SHA(0x002F), TLS_RSA_WITH_AES_256_CBC_SHA(0x0035), SSL_RSA_WITH_3DES_EDE_CBC_SHA(0x000A)]", 
  "compression methods" : "00", 
  "extensions"          : [ 
    "server_name (0)": { 
      type=host_name (0), value=build.mydomain.com 
    }, 
    "supported_groups (10)": { 
      "versions": [secp256r1, secp384r1, secp521r1, ffdhe2048, ffdhe3072, ffdhe4096, ffdhe6144, ffdhe8192] 
    }, 
    "ec_point_formats (11)": { 
      "formats": [uncompressed] 
    }, 
    "extended_master_secret (23)": { 
      <empty> 
    }, 
    "supported_versions (43)": { 
      "versions": [TLSv1] 
    }, 
    "renegotiation_info (65,281)": { 
      "renegotiated connection": [<no renegotiated connection>] 
    } 
  ] 
} 
) 
javax.net.ssl|FINE|4B|TC: 17:15:37 Build {id=1362, buildTypeId='BuildApplicationInfrastructureImages', buildTypeInternalId='bt10'}|2021-07-20 17:23:37.243 BST|Alert.java:238|Received alert message ( 
"Alert": { 
  "level"      : "fatal", 
  "description": "protocol_version" 
} 
) 
```

The plugin was failing at handshaking with the server because it found no common cipher suits to use. This is weird, because the client only offered a relatively short list of supported cipher suits to match.

Looking around in the log I found other handshakes that did not fail. These offered up many more supported cipher suits. Example:

```
javax.net.ssl|FINE|70|TC: 17:23:40 Task: 'jetbrains.buildServer.agent.impl.artifacts.ArtifactProcessor$5$1@605ae0f1'; Artifacts Processor pool 2|2021-07-20 17:23:40.351 BST|ClientHello.java:575|Produced ClientHello handshake message ( 
"ClientHello": { 
  "client version"      : "TLSv1.2", 
  "random"              : "5F DF DC 5B ED 1B 9F FD 59 5D 75 8C 22 5F E9 0A 13 93 C8 1C 41 02 83 23 EE 2B 7A 2F BE B6 59 FE", 
  "session id"          : "6D C7 91 54 66 32 09 6D 59 5D 1E 30 72 54 4E B1 7A F6 59 E4 B5 26 CC 7D 7A 53 D2 3D 66 80 E6 29", 
  "cipher suites"       : "[TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384(0xC02C), TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256(0xC02B), TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384(0xC030), TLS_RSA_WITH_AES_256_GCM_SHA384(0x009D), TLS_ECDH_ECDSA_WITH_AES_256_GCM_SHA384(0xC02E), TLS_ECDH_RSA_WITH_AES_256_GCM_SHA384(0xC032), TLS_DHE_RSA_WITH_AES_256_GCM_SHA384(0x009F), TLS_DHE_DSS_WITH_AES_256_GCM_SHA384(0x00A3), TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256(0xC02F), TLS_RSA_WITH_AES_128_GCM_SHA256(0x009C), TLS_ECDH_ECDSA_WITH_AES_128_GCM_SHA256(0xC02D), TLS_ECDH_RSA_WITH_AES_128_GCM_SHA256(0xC031), TLS_DHE_RSA_WITH_AES_128_GCM_SHA256(0x009E), TLS_DHE_DSS_WITH_AES_128_GCM_SHA256(0x00A2), TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384(0xC024), TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384(0xC028), TLS_RSA_WITH_AES_256_CBC_SHA256(0x003D), TLS_ECDH_ECDSA_WITH_AES_256_CBC_SHA384(0xC026), TLS_ECDH_RSA_WITH_AES_256_CBC_SHA384(0xC02A), TLS_DHE_RSA_WITH_AES_256_CBC_SHA256(0x006B), TLS_DHE_DSS_WITH_AES_256_CBC_SHA256(0x006A), TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA(0xC00A), TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA(0xC014), TLS_RSA_WITH_AES_256_CBC_SHA(0x0035), TLS_ECDH_ECDSA_WITH_AES_256_CBC_SHA(0xC005), TLS_ECDH_RSA_WITH_AES_256_CBC_SHA(0xC00F), TLS_DHE_RSA_WITH_AES_256_CBC_SHA(0x0039), TLS_DHE_DSS_WITH_AES_256_CBC_SHA(0x0038), TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256(0xC023), TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256(0xC027), TLS_RSA_WITH_AES_128_CBC_SHA256(0x003C), TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA256(0xC025), TLS_ECDH_RSA_WITH_AES_128_CBC_SHA256(0xC029), TLS_DHE_RSA_WITH_AES_128_CBC_SHA256(0x0067), TLS_DHE_DSS_WITH_AES_128_CBC_SHA256(0x0040), TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA(0xC009), TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA(0xC013), TLS_RSA_WITH_AES_128_CBC_SHA(0x002F), TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA(0xC004), TLS_ECDH_RSA_WITH_AES_128_CBC_SHA(0xC00E), TLS_DHE_RSA_WITH_AES_128_CBC_SHA(0x0033), TLS_DHE_DSS_WITH_AES_128_CBC_SHA(0x0032), TLS_ECDHE_ECDSA_WITH_3DES_EDE_CBC_SHA(0xC008), TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA(0xC012), SSL_RSA_WITH_3DES_EDE_CBC_SHA(0x000A), TLS_ECDH_ECDSA_WITH_3DES_EDE_CBC_SHA(0xC003), TLS_ECDH_RSA_WITH_3DES_EDE_CBC_SHA(0xC00D), SSL_DHE_RSA_WITH_3DES_EDE_CBC_SHA(0x0016), SSL_DHE_DSS_WITH_3DES_EDE_CBC_SHA(0x0013), TLS_EMPTY_RENEGOTIATION_INFO_SCSV(0x00FF)]", 
  "compression methods" : "00", 
  "extensions"          : [ 
    "server_name (0)": { 
      type=host_name (0), value=build.mydomain.com 
    }, 
    "supported_groups (10)": { 
      "versions": [secp256r1, secp384r1, secp521r1, ffdhe2048, ffdhe3072, ffdhe4096, ffdhe6144, ffdhe8192] 
    }, 
    "ec_point_formats (11)": { 
      "formats": [uncompressed] 
    }, 
    "signature_algorithms (13)": { 
      "signature schemes": [ecdsa_secp256r1_sha256, ecdsa_secp384r1_sha384, ecdsa_secp521r1_sha512, rsa_pss_rsae_sha256, rsa_pss_rsae_sha384, rsa_pss_rsae_sha512, rsa_pss_pss_sha256, rsa_pss_pss_sha384, rsa_pss_pss_sha512, rsa_pkcs1_sha256, rsa_pkcs1_sha384, rsa_pkcs1_sha512, dsa_sha256, ecdsa_sha224, rsa_sha224, dsa_sha224, ecdsa_sha1, rsa_pkcs1_sha1, dsa_sha1] 
    }, 
    "signature_algorithms_cert (50)": { 
      "signature schemes": [ecdsa_secp256r1_sha256, ecdsa_secp384r1_sha384, ecdsa_secp521r1_sha512, rsa_pss_rsae_sha256, rsa_pss_rsae_sha384, rsa_pss_rsae_sha512, rsa_pss_pss_sha256, rsa_pss_pss_sha384, rsa_pss_pss_sha512, rsa_pkcs1_sha256, rsa_pkcs1_sha384, rsa_pkcs1_sha512, dsa_sha256, ecdsa_sha224, rsa_sha224, dsa_sha224, ecdsa_sha1, rsa_pkcs1_sha1, dsa_sha1] 
    }, 
    "extended_master_secret (23)": { 
      <empty> 
    }, 
    "supported_versions (43)": { 
      "versions": [TLSv1.2, TLSv1.1, TLSv1] 
    } 
  ] 
} 
) 
javax.net.ssl|FINE|70|TC: 17:23:40 Task: 'jetbrains.buildServer.agent.impl.artifacts.ArtifactProcessor$5$1@605ae0f1'; Artifacts Processor pool 2|2021-07-20 17:23:40.353 BST|ServerHello.java:868|Consuming ServerHello handshake message ( 
"ServerHello": { 
  "server version"      : "TLSv1.2", 
  "random"              : "E6 22 9D 95 70 66 68 79 C5 6B 4C F1 1C 43 45 11 85 66 38 B5 49 F9 DC 12 44 4F 57 4E 47 52 44 01", 
  "session id"          : "A6 FE D3 12 FC EA 86 21 85 5B B0 F0 A3 05 C8 41 71 F1 B7 44 06 9F DA 42 53 F8 38 41 2A 08 91 15", 
  "cipher suite"        : "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384(0xC030)", 
  "compression methods" : "00", 
  "extensions"          : [ 
    "renegotiation_info (65,281)": { 
      "renegotiated connection": [<no renegotiated connection>] 
    }, 
    "server_name (0)": { 
      <empty extension_data field> 
    }, 
    "ec_point_formats (11)": { 
      "formats": [uncompressed, ansiX962_compressed_prime, ansiX962_compressed_char2] 
    }, 
    "extended_master_secret (23)": { 
      <empty> 
    } 
  ] 
} 
) 
```

I also compiled and ran Ciphers.java on the build agent (as described here) to verify that it indeed supported more ciphers and it generated the following output:

```
*       SSL_DHE_DSS_WITH_3DES_EDE_CBC_SHA
*       SSL_DHE_RSA_WITH_3DES_EDE_CBC_SHA
*       SSL_RSA_WITH_3DES_EDE_CBC_SHA
*       TLS_AES_128_GCM_SHA256
*       TLS_AES_256_GCM_SHA384
*       TLS_DHE_DSS_WITH_AES_128_CBC_SHA
*       TLS_DHE_DSS_WITH_AES_128_CBC_SHA256
*       TLS_DHE_DSS_WITH_AES_128_GCM_SHA256
*       TLS_DHE_DSS_WITH_AES_256_CBC_SHA
*       TLS_DHE_DSS_WITH_AES_256_CBC_SHA256
*       TLS_DHE_DSS_WITH_AES_256_GCM_SHA384
*       TLS_DHE_RSA_WITH_AES_128_CBC_SHA
*       TLS_DHE_RSA_WITH_AES_128_CBC_SHA256
*       TLS_DHE_RSA_WITH_AES_128_GCM_SHA256
*       TLS_DHE_RSA_WITH_AES_256_CBC_SHA
*       TLS_DHE_RSA_WITH_AES_256_CBC_SHA256
*       TLS_DHE_RSA_WITH_AES_256_GCM_SHA384
*       TLS_ECDHE_ECDSA_WITH_3DES_EDE_CBC_SHA
*       TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA
*       TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256
*       TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
*       TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA
*       TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384
*       TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
*       TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA
*       TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA
*       TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256
*       TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
*       TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA
*       TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384
*       TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
*       TLS_ECDH_ECDSA_WITH_3DES_EDE_CBC_SHA
*       TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA
*       TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA256
*       TLS_ECDH_ECDSA_WITH_AES_128_GCM_SHA256
*       TLS_ECDH_ECDSA_WITH_AES_256_CBC_SHA
*       TLS_ECDH_ECDSA_WITH_AES_256_CBC_SHA384
*       TLS_ECDH_ECDSA_WITH_AES_256_GCM_SHA384
*       TLS_ECDH_RSA_WITH_3DES_EDE_CBC_SHA
*       TLS_ECDH_RSA_WITH_AES_128_CBC_SHA
*       TLS_ECDH_RSA_WITH_AES_128_CBC_SHA256
*       TLS_ECDH_RSA_WITH_AES_128_GCM_SHA256
*       TLS_ECDH_RSA_WITH_AES_256_CBC_SHA
*       TLS_ECDH_RSA_WITH_AES_256_CBC_SHA384
*       TLS_ECDH_RSA_WITH_AES_256_GCM_SHA384
*       TLS_EMPTY_RENEGOTIATION_INFO_SCSV
        TLS_KRB5_WITH_3DES_EDE_CBC_MD5
        TLS_KRB5_WITH_3DES_EDE_CBC_SHA
*       TLS_RSA_WITH_AES_128_CBC_SHA
*       TLS_RSA_WITH_AES_128_CBC_SHA256
*       TLS_RSA_WITH_AES_128_GCM_SHA256
*       TLS_RSA_WITH_AES_256_CBC_SHA
*       TLS_RSA_WITH_AES_256_CBC_SHA256
*       TLS_RSA_WITH_AES_256_GCM_SHA384
```

So clearly the build agent can use other and stronger ciphers, but for some reason the REST-client used in this plugin limits itself to a few older and weaker ones. I don't know if this is a problem with the plugin, with the JetBrains REST-client used, Java-version or something else, but I'm probably not the only one who will experience this issue, so I'm hoping it can be resolved.

For now I got it working by weakening my Nginx config and allowing an additional cipher, but I would really like to avoid that:

```
ssl_ciphers ECDHE-RSA-AES256-GCM-SHA512:DHE-RSA-AES256-GCM-SHA512:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-SHA;
```

**Comment by rain-on** [comment]
Hi @olavaukan,
I'm just looking into this now.
It appears we have been using an old version of the teamcity-rest-client which is using an _old_ version of okhttp - I've updated the rest-client to its most recent version, but still need a test suite to see if its solved your issue.

**Comment by rain-on** [comment]
Hi @olavaukan,

The changes needed to update the HTTP client have been merged to master - you can build it using './gradlew distZip', then install 'build/distributions/Octopus.TeamCity.6.1.1-SNAPSHOT.zip' as a plugin to your TeamCity instance (and replace existing Octopus plugin).

If you're not in a position to build the plugin, email me at trent.mohay@octopus.com and I'll send through a zip file.

This change will be coming out as a formal release shortly, but it might take a few days.

**Comment by olavaukan** [comment]
Hi @rain-on,

Sorry for the slow response!

I haven't really touched Java since school 15+ years ago, so my attempt to build from source failed miserably:

```
> ./gradlew -Dteamcity.distribution=C:\TeamCity clean diztZip
Downloading https://services.gradle.org/distributions/gradle-7.1-bin.zip
..........10%...........20%...........30%..........40%...........50%...........60%..........70%...........80%...........90%...........100%

Welcome to Gradle 7.1!

Here are the highlights of this release:
 - Faster incremental Java compilation
 - Easier source set configuration in the Kotlin DSL

For more details see https://docs.gradle.org/7.1/release-notes.html

Starting a Gradle Daemon (subsequent builds will be faster)

FAILURE: Build failed with an exception.

* What went wrong:
Project '.distribution=C' not found in root project 'Octopus-Teamcity'.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 3m 16s
```

I'll take you up on that offer to send me the zip file ;-) Sending you an email.

**Comment by olavaukan** [comment]
I can confirm that this is fixed in 1.6.2. Thank you very much for the help @rain-on!
